packages/node/src/concordium/api.concordium.ts

```diff
@@ -0,0 +1,323 @@
+// Copyright 2020-2023 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {
+  AccountTransferredEvent,
+  BlockInfo,
+  ConcordiumGRPCClient,
+  TransactionEvent,
+  TransactionKindString,
+  TransactionSummaryType,
+  streamToList,
+} from '@concordium/node-sdk';
+import { ChannelCredentials } from '@grpc/grpc-js';
+import { EventEmitter2 } from '@nestjs/event-emitter';
+import { GrpcTransport, GrpcOptions } from '@protobuf-ts/grpc-transport';
+import { getLogger } from '@subql/node-core';
+import {
+  ApiWrapper,
+  ConcordiumBlock,
+  ConcordiumBlockWrapper,
+  ConcordiumSpecialEvent,
+  ConcordiumTransaction,
+  ConcordiumTransactionEvent,
+} from '@subql/types-concordium';
+import { flatMap, cloneDeep } from 'lodash';
+import { ConcordiumBlockWrapped } from './block.concordium';
+import SafeConcordiumGRPCClient from './safe-api';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { version: packageVersion } = require('../../package.json');
+
+const logger = getLogger('api.concordium');
+
+export class ConcordiumApi implements ApiWrapper<ConcordiumBlockWrapper> {
+  private client: ConcordiumGRPCClient;
+  private transport: GrpcTransport;
+  private genesisBlock: BlockInfo;
+  private chainId: string;
+  private name: string;
+
+  constructor(private endpoint: string, private eventEmitter: EventEmitter2) {
+    const options: GrpcOptions = {
+      host: this.endpoint,
+      channelCredentials: ChannelCredentials.createInsecure(),
+      meta: {
+        'User-Agent': `Subquery-Node ${packageVersion}`,
+      },
+    };
+
+    this.transport = new GrpcTransport(options);
+
+    this.client = new ConcordiumGRPCClient(this.transport);
+  }
+
+  async init(): Promise<void> {
+    const genesisBlock = await this.client
+      .getBlocksAtHeight(BigInt(0))
+      .then((hashes) => {
+        return this.client.getBlockInfo(hashes[0]);
+      });
+
+    this.genesisBlock = genesisBlock;
+    this.chainId = genesisBlock.blockHash;
+  }
+
+  async getFinalizedBlock(): Promise<BlockInfo> {
+    return this.client.getBlockInfo();
```

Comment:
Is there instant finalization with Concordium? If so can you please add a comment

Reply:
There is no instant finalization, I updated the getBestBlockHeight function to return the best block height